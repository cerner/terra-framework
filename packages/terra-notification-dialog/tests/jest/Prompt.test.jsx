import React from 'react';
import CompleteNotificationDialog from '../../examples/test-examples/CompleteNotificationDialog.example';
import CompleteNotificationDialogWithLongMessage from '../../examples/test-examples/CompleteNotificationDialogWithLongMessage.example';

it('should shallow a complete notification-dialog', () => {
  const modal = shallow(<CompleteNotificationDialog />);
  expect(modal).toMatchSnapshot();
});

it('should mount an complete notification-dialog', () => {
  const modal = mount(<CompleteNotificationDialog />);
  expect(modal).toMatchSnapshot();
});

it('should shallow a complete notification-dialog with long message', () => {
  const modal = shallow(<CompleteNotificationDialogWithLongMessage />);
  expect(modal).toMatchSnapshot();
});

it('should mount an complete notification-dialog with long message', () => {
  const modal = mount(<CompleteNotificationDialogWithLongMessage />);
  expect(modal).toMatchSnapshot();
});
