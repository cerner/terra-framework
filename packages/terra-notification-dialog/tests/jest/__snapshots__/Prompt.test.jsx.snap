// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount an complete notification-dialog 1`] = `
<CompleteNotificationDialog>
  <div>
    <NotificationDialog
      actions={
        Array [
          <Button
            id="notification-dialog-ok"
            isBlock={false}
            isCompact={false}
            isDisabled={false}
            isIconOnly={false}
            isReversed={false}
            onClick={[Function]}
            text="Ok"
            type="button"
            variant="neutral"
          />,
          <Button
            id="notification-dialog-close"
            isBlock={false}
            isCompact={false}
            isDisabled={false}
            isIconOnly={false}
            isReversed={false}
            onClick={[Function]}
            text="Close"
            type="button"
            variant="neutral"
          />,
        ]
      }
      message="Click ok to show alert and Close to go back."
      onRequestClose={[Function]}
      showNotificationDialog={false}
      title="Title goes here"
    />
    <Button
      id="trigger-notification-dialog"
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger NotificationDialog"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="button neutral"
        disabled={false}
        id="trigger-notification-dialog"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="text-only"
        >
          <span
            className=""
          >
            Trigger NotificationDialog
          </span>
        </span>
      </button>
    </Button>
  </div>
</CompleteNotificationDialog>
`;

exports[`should mount an complete notification-dialog with long message 1`] = `
<CompleteNotificationDialog>
  <div>
    <NotificationDialog
      actions={
        Array [
          <Button
            id="notification-dialog-ok"
            isBlock={false}
            isCompact={false}
            isDisabled={false}
            isIconOnly={false}
            isReversed={false}
            onClick={[Function]}
            text="Ok"
            type="button"
            variant="neutral"
          />,
          <Button
            id="notification-dialog-close"
            isBlock={false}
            isCompact={false}
            isDisabled={false}
            isIconOnly={false}
            isReversed={false}
            onClick={[Function]}
            text="Close"
            type="button"
            variant="neutral"
          />,
        ]
      }
      message="Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back."
      onRequestClose={[Function]}
      showNotificationDialog={false}
      title="Title goes here"
    />
    <Button
      id="trigger-notification-dialog"
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger NotificationDialog"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="button neutral"
        disabled={false}
        id="trigger-notification-dialog"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="text-only"
        >
          <span
            className=""
          >
            Trigger NotificationDialog
          </span>
        </span>
      </button>
    </Button>
  </div>
</CompleteNotificationDialog>
`;

exports[`should shallow a complete notification-dialog 1`] = `
<div>
  <NotificationDialog
    actions={
      Array [
        <Button
          id="notification-dialog-ok"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Ok"
          type="button"
          variant="neutral"
        />,
        <Button
          id="notification-dialog-close"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Close"
          type="button"
          variant="neutral"
        />,
      ]
    }
    message="Click ok to show alert and Close to go back."
    onRequestClose={[Function]}
    showNotificationDialog={false}
    title="Title goes here"
  />
  <Button
    id="trigger-notification-dialog"
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger NotificationDialog"
    type="button"
    variant="neutral"
  />
</div>
`;

exports[`should shallow a complete notification-dialog with long message 1`] = `
<div>
  <NotificationDialog
    actions={
      Array [
        <Button
          id="notification-dialog-ok"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Ok"
          type="button"
          variant="neutral"
        />,
        <Button
          id="notification-dialog-close"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Close"
          type="button"
          variant="neutral"
        />,
      ]
    }
    message="Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back.Click ok to show alert and Close to go back."
    onRequestClose={[Function]}
    showNotificationDialog={false}
    title="Title goes here"
  />
  <Button
    id="trigger-notification-dialog"
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger NotificationDialog"
    type="button"
    variant="neutral"
  />
</div>
`;
